python/core/qgsvectorlayer.sip

```diff
@@ -1202,6 +1202,42 @@ class QgsVectorLayer : QgsMapLayer
     // marked as const as these are just caches, and need to be created from const accessors
     void createJoinCaches() const;
 
+    /** Returns the calculated default value for the specified field index. The default
+     * value may be taken from a client side default value expression (see setDefaultValueExpression())
+     * or taken from the underlying data provider.
+     * @param index field index
+     * @param feature optional feature to use for default value evaluation. If passed,
+     * then properties from the feature (such as geometry) can be used when calculating
+     * the default value.
+     * @param context optional expression context to evaluate expressions again. If not
+     * specified, a default context will be created
+     * @return calculated default value
+     * @note added in QGIS 3.0
+     * @see setDefaultValueExpression()
+     */
+    QVariant defaultValue( int index, const QgsFeature& feature = QgsFeature(),
+                           QgsExpressionContext* context = nullptr ) const;
```

Comment:
I think so. It addresses a similar use case with a fraction more flexibility (if someone wanted to omit the project / global / layer scopes they could).